### PR TITLE
Resolve args.files before changing directory

### PR DESCRIPTION
**Pull request by bjorn-ove**
As discussed with @the-mikedavis in https://github.com/helix-editor/helix/issues/8520#issuecomment-1784158957

### CODE REVIEW

**Comment by pascalkuthe** [comment]
I think you can now simplify the logic to open the CWD again (get rid of the `open_cwd` flags and instead simply restore the logic to how it was before #8520)

**Comment by bjorn-ove** [comment]
Good point

**Comment by bjorn-ove** [comment]
One question that may be worth asking now is what is the logical behaviour of running `helix -w ~/.config ~/.ssh` or `helix -w ~/.config ~/.config/helix`

Currently it will effectively ignore `-w`, but perhaps it is better to change this, so it honors `-w` for working directory, and opens a picker at the first argument?

**Comment by the-mikedavis** [comment]
Yeah, I would prefer `-w` to take precedence, so we would set the cwd to `~/.config` in that first example and open a picker for `~/.ssh`

**Comment by bjorn-ove** [comment]
Ok, I made two changes now.

The first as discussed above, the second to open the remaining files in args.files when the first one is a directory.

File: helix-term/src/application.rs [hunk 1]
```diff
@@ -161,15 +161,20 @@ impl Application {
             editor.open(&path, Action::VerticalSplit)?;
             // Unset path to prevent accidentally saving to the original tutor file.
             doc_mut!(editor).set_path(None);
-        } else if !args.files.is_empty() {
-            if args.open_cwd {
-                // NOTE: The working directory is already set to args.files[0] in main()
-                editor.new_file(Action::VerticalSplit);
-                let picker = ui::file_picker(".".into(), &config.load().editor);
+        } else if let Some((first, _)) = args.files.first() {
+            let mut skip_files = 0;
```
Comment (the-mikedavis) [inline helix-term/src/application.rs:165]: It should be a little cleaner here to make an iterator over the files so we don't need to keep track of indices:

```rust
let files = args.files.iter(); // maybe `.peekable()` too if we want to check if the iterator is empty
// ..
if let Some((first, _)) = files.next() {
    // ..
}
```

Comment (bjorn-ove) [reply helix-term/src/application.rs:165]: I agree, just tried to keep refactoring to a minimum.

**Comment by bjorn-ove** [comment]
Updated after feedback

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**